frontend/src/main/java/de/iav/frontend/service/MetrologyService.java

```diff
@@ -0,0 +1,163 @@
+package de.iav.frontend.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.iav.frontend.model.Metrology;
+import de.iav.frontend.security.AuthenticationService;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class MetrologyService {
+
+    private static MetrologyService instance;
+    private final AuthenticationService authenticationService = AuthenticationService.getInstance();
+    private final HttpClient metrologyClient = HttpClient.newHttpClient();
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final String IAVCALIPRO_URL_BACKEND = System.getenv("BACKEND_IAVCALIPRO_URI");
+    private static final String JSON = "application/json";
+
+
+
+
+    public static synchronized MetrologyService getInstance() {
+
+        if (instance == null) {
+            instance = new MetrologyService();
+        }
+        return instance;
+    }
+
+    public List<Metrology> getMetrologyByMaintenanceOrCalibrationDue() {
+
+        List<Metrology> allMetrologies = getAllMetrologies();
+        List<Metrology> result = new ArrayList<>();
+
+        int maxDiff = 5;
+        Date currentDate = new Date();
+
+        for (Metrology metrology : allMetrologies) {
+
+            long maintenanceDiff = metrology.maintenance().getTime() - currentDate.getTime();
+            int maintenanceDiffDays = (int) (maintenanceDiff / (24 * 60 * 60 * 1000));
```

Comment:
Noch richtig Clean Code wäre vlt die Berechnung in eine Konstante auszulegen deren Name identifiziert was ihr Gebraucht ist.